lambdaclass/afk_gacha_game
# [GH-263] Fix tier up

Closes #263 

## Motivation

We weren't able to tier up our units once the max level of a tier was reached.

## Summary of changes

Add a tier up pop up with a button that allows tiering up the unit. Also, once the max tier of a rank has been reached, an error pop up is shown.

## How has this been tested?

1. Start the backend and add currency to your user by running: `GameBackend.Users.Currencies.add_currency_by_name_and_game!(user_id, "Gold", 2, 2000000)`
2. Go to the barracks and pick the unit you'll level up.
3. Level your unit up to level 20 and, when trying to level it up once more, make sure that you see the tier up pop up. 
4. Use the tier up button, and now you should be able to level up your unit once more. Check that the correct level is shown in the screen, and that the level up costs are being properly updated across this process.
5. Continue leveling and tiering up until you reach the max level of the max tier. When you try to tier up, you should now see an error pop up telling you that you must rank up your unit.

## Checklist
